src/concrete/ml/quantization/quantized_ops.py

```diff
@@ -174,19 +168,23 @@ def q_impl(
         alpha = self.attrs.get("alpha", 1)
         beta = self.attrs.get("beta", 1)
 
+        # If self.constant_inputs is empty this is an encrypted gemm
+        is_encrypted_gemm = isinstance(self.constant_inputs, dict) and not self.constant_inputs
+
         # If alpha != 1 or beta not in [0, 1], this function must be modified
         assert_true(alpha == 1)
-        assert_true(beta in [0, 1])
+        assert_true(beta in {0, 1})
 
         prepared_inputs = self._prepare_inputs_with_constants(
             *q_inputs, calibrate=False, quantize_actual_values=True
         )
 
-        q_input: QuantizedArray = prepared_inputs[0]
-        q_weights: QuantizedArray = prepared_inputs[1]
-        q_bias: Optional[QuantizedArray] = (
-            None if len(prepared_inputs) == 2 or beta == 0 else prepared_inputs[2]
-        )
+        q_input = prepared_inputs[0]
+        assert isinstance(q_input, QuantizedArray)
+        q_weights = prepared_inputs[1]
```

Comment:
the name "weights" works for training too but if we were to use this in a LLM we might want to change "weights" to "input2" no?

Reply:
Very good point. It's not weight anymore we should change this.

Reply:
called it input2 instead